### DISCUSSION TITLE
Troubleshooting 403 Errors with Load Balancer Configuration for Multiple Postal Console Domains

### DISCUSSION THREAD

**Discussion by emahdij** [postalserver/postal#3227] [Help with using Postal] [answered]
Hello everyone,

I have deployed two instances of Postal and configured the following domains:
- console.postal.com
- console2.postal.com

To manage traffic, I set up a load balancer that routes requests for `api.postal.com` to these servers. In the load balancer settings, I modified the Host header to match `console.postal.com` and `console2.postal.com`. Despite these configurations, I am encountering a 403 Forbidden error from the Postal web interface.

Here is my current Nginx configuration for console.postal.com:
```
server {
    listen 443 ssl;
    server_name console.postal.com;

    # SSL Configuration
    ssl_certificate /etc/letsencrypt/live/console.postal.com/fullchain.pem;
    ssl_certificate_key /etc/letsencrypt/live/console.postal.com/privkey.pem;

    location / {
        proxy_pass http://127.0.0.1:5000;
    }
}
```

This is the nginx log:
```
[11/Nov/2024:13:06:20 +0000] 94.101.182.8 - - "GET / HTTP/1.1" 403 0 "-" "curl/8.7.1" "37.32.9.19" Header-Host: console.postal.com, User-Agent: curl/8.7.1, Request-Time: 0.006, Upstream-Time: 0.006, Request-URI: /, Cookies: "-", Response-Content-Type: text/html; charset=UTF-8
```

How can I resolve the 403 Forbidden error when accessing the Postal web interface through the load balancer? or
Is it possible to configure multiple domains for the Postal console, or is this limitation inherent to Postal?

Any guidance or suggestions would be greatly appreciated!
Thank you!

**Comment by willpower232** [comment]
I think that Rails, the underlying framework, is quite strict about which hostnames it answers to. What did you set for `web_hostname` in the config file or environment?

**Reply by emahdij** [reply]
I set console.postal.com and console2.postal.com. Panel open with these two domains.

**Reply by willpower232** [reply]
I thought the config only takes one domain so not sure what happened but I presume you can "cheat" by overriding the host header in the location block to match the domain set in the config file.

Otherwise you would have to modify this file and make your own docker image
https://github.com/postalserver/postal/blob/fd3c7ccdf6dc4ee0a76c9523cbd735159e4b8000/config/application.rb#L43

**Reply by emahdij** [reply]
The problem is I've changed my host header from api.postal.com ro console.postal.com in the loadbalancer. You can find it in nginx log. My problem is how the web server realize and wich parameter is important for it to change.

**Comment by emahdij** [comment, marked as answer]
I fixed this problem with Nginx configuration. I added these two lines:
proxy_set_header Host $host;
proxy_set_header X-Forwarded-Host $host;
